projects/packages/search/src/classic-search/class-classic-search.php

```diff
@@ -95,29 +95,26 @@ protected function __construct() {
 	}
 
 	/**
-	 * Returns the singleton of the class. Does not perform any instantiation.
-	 *
-	 * @return static The class singleton.
-	 */
-	public static function instance() {
-		return self::$instance;
-	}
-
-	/**
-	 * Instantiate and initialize a singleton instance of the class.
+	 * Returns a class singleton. Initializes with first-time setup if given a blog ID parameter.
 	 *
 	 * @param string $blog_id Blog id.
 	 * @return static The class singleton.
 	 */
-	public static function initialize( $blog_id ) {
-		if ( ! isset( self::$instance ) ) {
+	public static function instance( $blog_id = null ) {
```

Comment:
Could we set `blog_id` to the current blog id inside the initializing process as a fallback? Having `$blog_id` as optional parameter is compatible with the old versions however would still possibly return a un-initialized instance if not used properly.

Reply:
Good point, I'll implement this change.